docs/sources/get-started/components.md

```diff
@@ -217,24 +252,87 @@ The query frontend splits larger queries into multiple smaller queries, executin
 
 ### Caching
 
-#### Metric Queries
+#### Metric queries
+
+The query frontend supports caching metric query results and reuses them on subsequent queries. If the cached results are incomplete, the query frontend calculates the required sub-queries and executes them in parallel on downstream queriers. The query frontend can optionally align queries with their step parameter to improve the cacheability of the query results. The result cache is compatible with any Loki caching backend (currently Memcached, Redis, and an in-memory cache).
+
+#### Log queries
+
+The query frontend also supports caching of log queries in form of a negative cache.
+This means that instead of caching the log results for quantized time ranges, Loki only caches empty results for quantized time ranges.
+This is more efficient than caching actual results, because log queries are limited (usually 1000 results)
+and if you have a query over a long time range that matches only a few lines, and you only cache actual results,
+you'd still need to process a lot of data additionally to the data from the results cache in order to verify that nothing else matches.
+
+#### Index stats queries
+
+The query frontend caches index stats query results similar to the [metric query](#metrics-queries) results.
+This cache is only applicable when using single store TSDB.
+
+#### Log volume queries
 
-The query frontend supports caching metric query results and reuses them on subsequent queries. If the cached results are incomplete, the query frontend calculates the required subqueries and executes them in parallel on downstream queriers. The query frontend can optionally align queries with their step parameter to improve the cacheability of the query results. The result cache is compatible with any Loki caching backend (currently memcached, redis, and an in-memory cache).
+The query frontend caches log volume query results similar to the [metric query](#metrics-queries) results.
+This cache is only applicable when using single store TSDB.
 
-#### Log Queries - Coming soon!
 
-Caching log (filter, regexp) queries are under active development.
+## Query scheduler
+
+The **query scheduler** is an **optional service** providing more [advanced queuing functionality]({{< relref "../operations/query-fairness" >}}) than the [query frontend](#query-frontend).
+When using this component in in the Loki deployment, query frontend pushes split up queries to the query scheduler which enqueues them in an internal in-memory queue.
+There is a queue for each tenant to guarantee the query fairness across all tenants.
+The queriers that connect to the query scheduler act as workers that pull their jobs from the queue, execute them, and return them to the query-frontend for aggregation. Queriers need to be configured with the query frontend address (via the `-querier.frontend-address` CLI flag) in order to allow them to connect to the query frontends.
```

Comment:
should this be corrected to use scheduler address when query scheduler is deployed?

Reply:
Good catch!